sourceme.sh

```diff
@@ -1,10 +1,17 @@
-#!/bin/sh
+#!/bin/bash
 # Convenience script for setting variables for in-tree use of hammer.
-if [ -z "$HAMMER_HOME" ]; then
-    >&2 echo "Must set HAMMER_HOME to root. Try export HAMMER_HOME="\$PWD" for the current dir."
+
+if [ -z "${HAMMER_HOME}" ]; then
+  # Try to find the location of the script even if it's called through a symlink.
+  #  https://stackoverflow.com/questions/59895/how-to-get-the-source-directory-of-a-bash-script-from-within-the-script-itself
+  #  https://stackoverflow.com/questions/35006457/choosing-between-0-and-bash-source
+  export HAMMER_HOME="$( dirname "$( readlink -f "${BASH_SOURCE[0]}" )" )"
+  >&2 echo "Using (guessed) HAMMER_HOME=${HAMMER_HOME}"
```

Comment:
I'm not sure I'd add the (guessed)/(existing) parts here because it might actually confuse people. In this case if you source this more than once, it will tell you that it's using an (existing) HAMMER_HOME the second time because you set it the first time. IMO, a message like this would be a little more explanatory.
```suggestion
  >&2 echo "Setting HAMMER_HOME to ${HAMMER_HOME}"
```

Reply:
done